### ISSUE TITLE
Bpod_ttl_test Missing Tones

### ISSUE THREAD

**Issue by ms2565** [int-brain-lab/iblrig#439]
On running IBL->Experiments->_iblrig_misc->bpod_ttl_test, (immediately after initializing the iblrig GUI using 'start-pybpod' terminal command):

The frame updates and go tones don't seem to be synchronized as the prompt output reads (for each starting trial):

2023-02-03 13:15:14.812 INFO     [frame2TTL.py:462] Setting dark threshold to -344
2023-02-03 13:15:14.813 INFO     [frame2TTL.py:464] Setting light threshold to 186
2023-02-03 13:15:14.814 INFO     [frame2TTL.py:498] Frame2TTL: Thresholds set.
2023-02-03 13:15:18.505 INFO     [_iblrig_misc_bpod_ttl_test.py:65] Starting trial: 1
**2023-02-03 13:15:26.765 WARNING  [_iblrig_misc_bpod_ttl_test.py:141] Missing tones: 1**
2023-02-03 13:15:26.766 INFO     [_iblrig_misc_bpod_ttl_test.py:65] Starting trial: 2
**2023-02-03 13:15:27.901 WARNING  [_iblrig_misc_bpod_ttl_test.py:141] Missing tones: 2**
...
...
2023-02-03 13:15:54.417 ERROR    [session_params.py:128]
        ##########################################
                      TEST FAILED !!!!!
        ##########################################
          Missing stimulus sync pulses: {self.bad_stim_count}
          Missing tone sync pulses:     {self.bad_tone_count}
        ##########################################

**To Reproduce**

1. Open Windows PowerShell (Microsoft.PowerShell_7.3.2.0_x64...)
2. Run the following commands:
- C:\iblrig\venv\Scripts\.\Activate.ps1
- C:\iblrig_params
- start-pybpod
3. 'Double-click' user (e.g. '_iblrig_test_user') under 'Users'
4. Load board Info
- Click 'Bpod boards'->'_iblrig_testlab_behavior_1'
- Click (now on RHS of screen, under 'Details') 'Load board info'
5. Run bpod_ttl_test
- Click '_iblrig_misc'->'bpod_ttl_test'
- Click (on RHS, under 'Details') 'Run'

**Expected behavior**
I expect in the output terminal to not read any warning lines describing 'Missing tones' for any of the 'Starting trials' (e.g.):

2023-02-03 13:15:14.812 INFO     [frame2TTL.py:462] Setting dark threshold to -344
2023-02-03 13:15:14.813 INFO     [frame2TTL.py:464] Setting light threshold to 186
2023-02-03 13:15:14.814 INFO     [frame2TTL.py:498] Frame2TTL: Thresholds set.
2023-02-03 13:15:18.505 INFO     [_iblrig_misc_bpod_ttl_test.py:65] Starting trial: 1
2023-02-03 13:15:26.766 INFO     [_iblrig_misc_bpod_ttl_test.py:65] Starting trial: 2
...
...
2023-02-03 <*some time stamp later*> INFO     [session_params.py:109] Checking...
2023-02-03 <*some time stamp later*> INFO     [session_params.py:109]
#########################################
                              TEST PASSED:
#########################################
             Stimulus and tone detected in all trials
#########################################

**Screenshots**
![image](https://user-images.githubusercontent.com/108840201/216684566-7c0d9059-97d4-4e72-ba13-c4ba12d047f5.png)

![image](https://user-images.githubusercontent.com/108840201/216684642-59663d8c-aa50-4eaa-b7bd-2f97c349cea9.png)

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Windows 
 - Browser [e.g. chrome, safari] Chrome
 - Version [e.g. 22] Windows 10

**Additional context**
We're currently unable to obtain a luxmeter to calibrate the proper Frame2TTL thresholds so, prior to running 'bpod_ttl_test' I'm running the frame2TTL calibration without a polarizing filter (I'm wondering if this might be obscuring the frame to go tone synchronization)

I'm currently reading through the source code in the iblrig file structures to see if I can pinpoint the error.

Any help/suggestions/explanations about what is going on (in the backend) when I try to run this Hardware Connection Test would be greatly appreciated. Please let me know if there's any more information I can provide.

**Comment by jefferal1995** [comment]
@ms2565, do you know what the issue was? I am having the same problem. I am not sure if you were able to solve it

**Comment by ms2565** [comment]
Hey @jefferal1995 ! Yes, I traced the bpod_ttl_test.py workflow and figured out how the state machine transitions from one state to the next (I put together a diagram that helped me visualize, see below). I found, in my case, that the bad_tone state (i.e. the state associated with the output you might be receiving at each trial: 'Missing tones: <some number>') is coupled to the TTL signal present on the BNC2 cable plugged into the bpod module.

Whenever the bpod_ttl_test state machine enters the 'play_tone' state, it can either transition to the 'exit' state (at which point it will re-initiate and start the next trial) or—if it doesn't properly receive the TTL from BNC2—it will transition to the 'bad_tone' state and give you the 'missing tones' output.

The issue resolved when I gently jostled the BNC2 cable in the back of my iblrig setup and fixed the mechanical connection (e.g. kind of like how your old wired headphones, back in the day, maybe only worked if you held/pressed them into your phone/ipod jack at the right angle).

I might recommend you check the quality of your BNC2 cable or the physical point of connection or otherwise more deeply investigate factors that might be contributing to a breakdown in synchrony between the TTL pulses.

Hope that helps, at least with getting started on the issue! Let me know if anything I mentioned could use more clarifying/details.

Best of luck!
-Marc
![image](https://user-images.githubusercontent.com/108840201/222786269-f4eb9639-4cf5-4502-ae56-310ece69b883.png)
